packages/promises/_test.pony

```diff
@@ -145,3 +146,22 @@ class iso _TestPromisesJoinThenReject is UnitTest
     a("a")
     b("b")
     c.reject()
+
+class iso _TestNextUnwrap is UnitTest
+  fun name(): String => "promises/Promise.flatten_next"
+
+  fun apply(h: TestHelper) =>
+    let start = Promise[String]
+    let inter = start.flatten_next[String](_NextUnwrapHelper~promise_string())
+    inter.next[None](_NextUnwrapHelper~complete(h))
+
+    start("start")
+    inter("end")
+
```

Comment:
I'd like to make sure we test a tiny bit more aggressively here

* Create a promise, apply it to a value + Return it should call apply with that value.
* Create a promise and reject it. Then the outermost promise should be rejected.
* Error out when creating a promise, Then the outermost promise should be rejected

Reply:
Resolving to clear up the comments